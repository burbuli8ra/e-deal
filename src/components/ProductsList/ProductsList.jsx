import React from 'react';
import { ProductBlock } from 'components';
import { useAppContext } from 'provider';

const ProductsList = ({ products }) => {
  const [{ error }] = useAppContext();

  return (
    <>
      {error || products.map(product =>
        <ProductBlock
          imageUrl={product.image_url}
          key={`${product.id}-${product.slug_path}`}
          {...product}
        />
      )}
    </>
  );
};

export default ProductsList;