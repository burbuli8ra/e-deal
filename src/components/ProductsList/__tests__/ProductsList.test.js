import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { fireEvent, render } from '@testing-library/react';
import { AppProvider } from 'provider';
import Mocks from '__mocks__';
import ProductsList from '../ProductsList';

const setup = props => (
  <AppProvider>
    <Router>
      <ProductsList {...props} />
    </Router>
  </AppProvider>
);

describe('ProductsList component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      products: Mocks.ProductsList
    };
    wrapper = render(setup(props));
  });

  test('should match snapshot', () => {
    const { asFragment, getAllByTestId } = wrapper;

    getAllByTestId('productItem-ghost-image').forEach(img =>
      fireEvent.load(img)
    );

    expect(asFragment()).toMatchSnapshot();
  });
});