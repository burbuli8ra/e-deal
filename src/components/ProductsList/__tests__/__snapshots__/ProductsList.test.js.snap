// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList component should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2155431109"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Hobby HSM-1000 White"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg34/2vgIi/1JiA8P_SX910Y902/0/2155431109.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Hobby HSM-1000 White
    </h3>
    <div
      class="css-g0fl43"
    >
      61.90€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2155402049"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Hobby HSM-1000 Grey"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg33/2vgIh/1JiA8P_SX918Y889/0/2155402049.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Hobby HSM-1000 Grey
    </h3>
    <div
      class="css-g0fl43"
    >
      62.00€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2155062600"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Pyrex SB-1000 Gold"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg31/1TgXZ/1JcqHd_SX585Y639/0/2155062600.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Pyrex SB-1000 Gold
    </h3>
    <div
      class="css-g0fl43"
    >
      66.90€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2154612159"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Human TM-617"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg33/31Wxj/1Jj9Vj_SX354Y366/0/2154612159.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Human TM-617
    </h3>
    <div
      class="css-g0fl43"
    >
      68.90€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2154682235"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Royalty Line RL-PKM1400.5 Red"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg59/2lOPhh/1GngAT_SX508Y512/0/2154682235.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Royalty Line RL-PKM1400.5 Red
    </h3>
    <div
      class="css-g0fl43"
    >
      69.79€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2155445799"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Life KM-001"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg19/2HT6X/1JgsPL_SX911Y750/0/2155445799.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Life KM-001
    </h3>
    <div
      class="css-g0fl43"
    >
      76.90€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2154682236"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Royalty Line RL-PKM1400.5 Black"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg44/3mlm4/1JkM8F_SX928Y934/0/2154682236.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Royalty Line RL-PKM1400.5 Black
    </h3>
    <div
      class="css-g0fl43"
    >
      77.30€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2153792366"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Gruppe OU 6119 Red"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg33/2YpkB/1JmsSL_SX578Y510/0/2153792366.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Gruppe OU 6119 Red
    </h3>
    <div
      class="css-g0fl43"
    >
      78.99€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2155877685"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Izzy Spicy Red 800"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg37/3sB7f/1JcqHd_SX681Y639/0/2155877685.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Izzy Spicy Red 800
    </h3>
    <div
      class="css-g0fl43"
    >
      79.00€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2155738384"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Royalty Line RL-PKM1600W-BLK"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg21/3930x/1JgKCf_SX833Y822/0/2155738384.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Royalty Line RL-PKM1600W-BLK
    </h3>
    <div
      class="css-g0fl43"
    >
      84.84€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2155168476"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Royalty Line RL-PKM1900.7 Red"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg28/2lQRLS/1Jhs8w_SX838Y996/0/2155168476.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Royalty Line RL-PKM1900.7 Red
    </h3>
    <div
      class="css-g0fl43"
    >
      86.90€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2155168477"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Royalty Line RL-PKM1900.7 Black"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg25/3CO5r/1Jmrx3_SX889Y990/0/2155168477.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Royalty Line RL-PKM1900.7 Black
    </h3>
    <div
      class="css-g0fl43"
    >
      86.90€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2155431104"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Izzy Super Chef S1503"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg0/2lRY5O/1Jhs7e_SX641Y697/0/2155431104.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Izzy Super Chef S1503
    </h3>
    <div
      class="css-g0fl43"
    >
      92.00€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2153132249"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="First Austria FA-5259"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg25/2lIk3v/1FjDz4_SX462Y367/0/2153132249.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      First Austria FA-5259
    </h3>
    <div
      class="css-g0fl43"
    >
      98.00€
    </div>
  </a>
  <a
    class="css-1sj3ugd"
    data-testid="productItem-block"
    href="/products/2154790570"
  >
    <div
      class="css-1yv013g"
    >
      <img
        alt="Sencor STM 3625VT"
        class="css-y6bwx0"
        src="https://abpcdn.pstatic.gr/P/bpimg2/30XN8/1Jmd2W_SX1157Y1200/0/2154790570.webp"
      />
    </div>
    <h3
      class="css-15l8jez"
    >
      Sencor STM 3625VT
    </h3>
    <div
      class="css-g0fl43"
    >
      99.00€
    </div>
  </a>
</DocumentFragment>
`;
